tests/test_try.py

```diff
@@ -0,0 +1,209 @@
+    )
+    assert response.status_code == 400, "Try push without patches should return 400."
+
+
+def test_try_api_success_hgexport(
+    app,
+    db,
+    hg_server,
+    hg_clone,
+    treestatusdouble,
+    client,
+    auth0_mock,
+    mocked_repo_config,
+):
+    treestatus = treestatusdouble.get_treestatus_client()
+    treestatusdouble.open_tree("mozilla-central")
+
+    try_push_json = {
+        # The only node in the test repo.
+        "base_commit": "0da79df0ffff88e0ad6fa3e27508bcf5b2f2cec4",
+        "patch_format": "hgexport",
+        "patches": [
+            base64.b64encode(PATCH_WITHOUT_STARTLINE).decode("ascii"),
+        ],
+    }
+    response = client.post(
+        "/try/patches", json=try_push_json, headers=auth0_mock.mock_headers
+    )
+    assert response.status_code == 201, "Successful try push should return 201."
+    assert (
+        "id" in response.json
+    ), "Response should include the ID of the new landing job."
+
+    queue_items = LandingJob.job_queue_query(
+        repositories=["try"], grace_seconds=0
+    ).all()
+    assert len(queue_items) == 1, "Try push should have created 1 landing job."
+
+    # Run the landing job.
+    job = queue_items[0]
+
+    repo = Repo(
+        tree="try",
+        url=hg_server,
+        access_group=SCM_LEVEL_1,
+        push_path=hg_server,
+        pull_path=hg_server,
+        short_name="try",
+        is_phabricator_repo=False,
+        force_push=True,
+    )
+
+    worker = LandingWorker(sleep_seconds=0.01)
+    hgrepo = HgRepo(hg_clone.strpath)
+
+    assert worker.run_job(job, repo, hgrepo, treestatus)
+    assert job.status == LandingJobStatus.LANDED
+    assert len(job.landed_commit_id) == 40
+    assert (
+        job.target_commit_hash == "0da79df0ffff88e0ad6fa3e27508bcf5b2f2cec4"
+    ), "Target changeset should match the passed value."
+
+
+def test_try_api_success_gitformatpatch(
+    app,
+    db,
+    hg_server,
+    hg_clone,
+    treestatusdouble,
+    client,
+    auth0_mock,
+    mocked_repo_config,
+):
+    treestatus = treestatusdouble.get_treestatus_client()
+    treestatusdouble.open_tree("mozilla-central")
+
+    try_push_json = {
+        # The only node in the test repo.
+        "base_commit": "0da79df0ffff88e0ad6fa3e27508bcf5b2f2cec4",
+        "patch_format": "git-format-patch",
+        "patches": [
+            base64.b64encode(GIT_PATCH).decode("ascii"),
+        ],
+    }
+    response = client.post(
+        "/try/patches", json=try_push_json, headers=auth0_mock.mock_headers
+    )
+    assert response.status_code == 201, "Successful try push should return 201."
+    assert (
+        "id" in response.json
+    ), "Response should include the ID of the new landing job."
+
+    queue_items = LandingJob.job_queue_query(
+        repositories=["try"], grace_seconds=0
+    ).all()
+    assert len(queue_items) == 1, "Try push should have created 1 landing job."
+
+    # Run the landing job.
+    job = queue_items[0]
```

Comment:
Can you be more specific about what you think should be checked?

Reply:
Checking `job.revisions` length, that revisions were created as expected (this would be coverage for `Revision.new_from_patch`), etc... This would also help people reading this integration test understand the code flow.